### PR TITLE
Pass down the feature flag parameter for the task list in Payments > Overview

### LINKED ISSUE #4471: Not all tasks in the Payments → Overview task list are visible
### Describe the bug

The **Payments → Overview** `Things to do` task list should render the [following tasks](https://github.com/Automattic/woocommerce-payments/blob/37997a7e550005c1d1ca827e6ffbf8da7b15093e/client/overview/task-list/tasks.js#L65) when the [`_wcpay_feature_account_overview_task_list`](https://github.com/Automattic/woocommerce-payments/blob/2f0ffe9d3e91b4a037bc5e4636b01ad757bf2985/includes/class-wc-payments-features.php#L60-L62) feature flag is set to `'1'`.

The **Payments → Overview** `Things to do` task list is set for launch with the release of WCPay 4.5.0 #4460.

* **⚠️ Not visible:** [`update-business-details`](https://github.com/Automattic/woocommerce-payments/pull/1799): Update WooCommerce Payments business details.
* **⚠️ Not visible:** [`reconnect-wpcom-user`](https://github.com/Automattic/woocommerce-payments/pull/1912): WooCommerce Payments is missing a connected WordPress.com account. Some functionality will be limited without a connected account.
* **⚠️ Not visible:** [`force-secure-checkout`](https://github.com/Automattic/woocommerce-payments/pull/2480): Protect your customers data and increase trustworthiness of your store by forcing HTTPS on checkout pages.
* **Visible:** `dispute-resolution-task`: *x* disputed payments needs your response.

> **Warning**
> The tasks that are currently **not visible** should be reviewed from a design/product perspective before fixing this issue. See discussion: paJDYF-4EQ-p2#comment-12661

### To Reproduce

1. Meet the criteria for each task above to be visible. The simplest way is to overwrite each setting in [`class-wc-payments-admin.php`→`register_payments_scripts()`→`$wcpay_settings`](https://github.com/Automattic/woocommerce-payments/blob/55916cc468e361202601981767a81ddea9456fc9/includes/admin/class-wc-payments-admin.php#L452) to the following:
      * `update-business-details`: `'showUpdateDetailsTask' => 'yes'`
      * `reconnect-wpcom-user`: `'wpcomReconnectUrl' => WC_Payments_Account::get_wpcom_reconnect_url()`
      * `force-secure-checkout`: `'needsHttpsSetup' => 'yes'`
      * `dispute-resolution-task`: `'numDisputesNeedingResponse' => 10`
1. Visit **Payments → Overview** in wp-admin. Notice that only the `dispute-resolution-task` is visible
   * ⚠️ only in WCPay v4.5.0+ from which the `_wcpay_feature_account_overview_task_list` feature flag defaults to `'1'`.

### Actual behavior

Only the `dispute-resolution-task` is rendered.

image

### Expected behavior

All tasks listed above are visible.

image

### Additional context

Setting the default value `isAccountOverviewTasksEnabled = true` in the [`getTasks()` function](https://github.com/Automattic/woocommerce-payments/blob/55916cc468e361202601981767a81ddea9456fc9/client/overview/task-list/tasks.js#L20) allows all list items to be displayed, so it looks like the feature flag is not being passed through to this function correctly.

Slack discussion: p1658460988099129-slack-C03LA109BKM
P2 discussion about design/product revision of task list items: paJDYF-4EQ-p2#comment-12661

### PR DESCRIPTION
**Pull request by jbordonado**
Fixes #4471 

#### Changes proposed in this Pull Request

The task list feature flag was mistakenly omitted from the Tasks component for the Payments > Overview page. Consequently, we couldn't see 3 tasks even though that feature flag is now active (planned to be removed in #4489).
Everything works out of the box once the feature flag parameter is passed down to the Tasks component.

**One thing to note**: since the `Force secure checkout` task also exists as an inbox note in WooCommerce Home, it was decided to remove that task from the Payments > Overview page and reword the related note so that we use "Enable" (see the reasoning here: https://github.com/Automattic/woocommerce-payments/pull/4707#issuecomment-1241118531)

#### Testing instructions

To enable the 2 tasks, follow the below instructions:

- Update WooCommerce Payments business details 

Hardcode the value `'yes'` at the following line: https://github.com/Automattic/woocommerce-payments/blob/f2a1437c69dcfa7760a98e653b34d6ddfdab6739/includes/admin/class-wc-payments-admin.php#L474

- WooCommerce Payments is missing a connected WordPress.com account. Some functionality will be limited without a connected account.
 
Hardcode the value `'WC_Payments_Account::get_wpcom_reconnect_url()'` at the following line: https://github.com/Automattic/woocommerce-payments/blob/f2a1437c69dcfa7760a98e653b34d6ddfdab6739/includes/admin/class-wc-payments-admin.php#L475

To see the new wording for the "Secure checkout" inbox note, you should just need to deactivate and activate again the WCPay plugin.

Once that is done, you should:
- see those 2 tasks in the "Things to do" when visiting the Payments > Overview page and also notice that we're using the word "Enable" instead of "Force" in the WooCommerce inbox note, if you're using that branch
- not see any of those tasks, and still see "Force secure checkout" for the inbox note if you're using the `develop` branch

Payments > Overview "Things to do" list
image

WooCommerce Home > Inbox note
image

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by jbordonado** [comment]
@elizaan36 Do we need to adjust the wording/design for any of the three tasks (see screenshot below)? I see in the issue that:

> The tasks that are currently not visible should be reviewed from a design/product perspective

image

Once that PR is merged, those tasks will be visible for merchants if the context allows it, so any design change needs to happen before we do that.

**Comment by elizaan36** [comment]
hi @jbordonado 👋 Thanks for the ping.

I have been wanting to tweak that task, "Force secure checkout" since it won't be super clear to merchants. Could we change the copy to:
> **Enable secure checkout**
Enable HTTPS on your checkout pages to display all available payment methods and protect your customers data.

For "Reconnect WooCommerce Payments" and "Force secure checkout", is the merchants redirected away from this screen? How long does the process take in the background?

I'm wondering if you could check something for me as well, as I'm still in the process of setting up my local dev. Could you demo the ellipses menu (on the right side of the task) and share a quick gif on here? There's an issue with the dropdown being hidden on the bottom task in the menu and wonder if we could fix that as part of this work. [Issue created here](https://github.com/Automattic/woocommerce-payments/issues/4470).

**Comment by jbordonado** [comment]
@elizaan36 

> Force secure checkout

Clicking on that task opens a new tab linking to https://woocommerce.com/document/ssl-and-https/#section-7, which takes 1 or 2s on my machine.

Another remark on that topic: we have two other places where we use this wording.

- There is one note that WCPay adds to WooCommerce admin, and we can also adjust the wording:
image

And there is a setting in WooCommerce > Settings > Advanced, which is built within Woo core repository, not here:
image

So do you think it should be aligned in those 3 places? I can align the wording in the Woo admin note and the Payments task list, but it'll require another dev in Woo core to align the 3rd place (the advanced setting).

> Reconnect WooCommerce Payments

Clicking on the task will collapse/expand the task. Inside the task, there's a "Reconnect" button. If you click on it, then you are redirected to the Jetpack connect screen after a few seconds: 
image

Clicking on "Approve" will take a bit more time, like 20 or 30s, and then you land on the WooCommerce > Settings screen with WooCommerce Payments gateway selected (so you see the WCPay settings):
image

For that task, just noting here that I had to disconnect Jetpack from my site to be in the right context before trying the button. Once this is done, visiting Payments > Home or Payments > Overview will leave me stuck on the home screen, asking me to "Finish setup." I couldn't see a way to land on the Payments > Overview screen unless you had another tab opened like I did (to see what happens when I click on the "Reconnect" button).

> Could you demo the ellipses menu (on the right side of the task) and share a quick gif on here? There's an issue with the dropdown being hidden on the bottom task in the menu and wonder if we could fix that as part of this work. https://github.com/Automattic/woocommerce-payments/issues/4470.

Sept-08-2022 14-51-04

I'm not experiencing the issue described in #4470. I'm using Chrome v105 on macOS v12.3.1 FYI
I also noticed that when you created that issue, the "Things to do" list was just above the notes.
In my case, there is the "Account details" section just below (and then the notes), so that's maybe why I'm not reproducing the issue. I guess there's been a change merged to move the list on top. Maybe the problem was with the "notes" section, and it's now separated from the "Things to do".
Can you revisit the issue and confirm this is no longer reproducible? We could close it then.

**Comment by elizaan36** [comment]
> There is one note that WCPay adds to WooCommerce admin

There's no reason to duplicate this information in the two places, it's a lot of noise for merchants. 

Considering that the note takes you to documentation, it's more of an informational nature. Tasks should direct users to take action, so I don't think we should implement this as a task in this case. If the task was going to direct the user to Settings, or update the settings for the user upon clicking it, then I think it should be a task.

IMO the task should be removed in favor of the Inbox note (with updated language).

> it'll require another dev in Woo core to align the 3rd place (the advanced setting)

I would change the word "Force" to "Enable" in Advanced settings but it's not a super high priority.

> For that task, just noting here that I had to disconnect Jetpack from my site to be in the right context before trying the button. Once this is done, visiting Payments > Home or Payments > Overview will leave me stuck on the home screen, asking me to "Finish setup." I couldn't see a way to land on the Payments > Overview screen unless you had another tab opened like I did (to see what happens when I click on the "Reconnect" button).

I've seen this feedback from merchants in the new user survey. This sounds like a huge pain - Do you think it's worth creating a new issue for this?

Thanks for testing the dismiss action! Nice to see that it can't be reproduced on your end, I updated #4470.

**Comment by jbordonado** [comment]
To recap, after merging this PR:

- We could now show two additional tasks in the 'Things to do" list from Payments > Overview
   - One task is about updating the business details
   - One task is about reconnecting WooCommerce Payments
- The note about "force secure checkout" was reworded as demanded

The new potential tasks in Payments > Overview:
image

The WooCommerce > Home note about secure checkout:
image

If all is fine @elizaan36, I will ask to get this reviewed, and it should ship with the next release (WCPay 4.8.0 - scheduled to be released on Sept 28).

Regarding the advanced settings, would you mind opening a low-priority issue to get the rewording sorted in [Woo admin](https://github.com/woocommerce/woocommerce)?

As for the feedback you've seen in the new user survey, could you share more details? I was just stating that it seemed unclear to me how one merchant could ever see the task "Reconnect WCPay" given that we seemed to require to be connected to even reach that Payments > Overview screen. But there might be other ways to reach that state that I'm not familiar with.
Are you referring to things like merchants having already set up a WCPay account yet still seeing a "Finish setup" screen when trying to access the Payments page? Those are quite complicated to investigate as we often don't have a clear way to reproduce the behavior. I know there have been some fixes shipped already regarding those kinds of issues, but there could remain problems, yes.
If there is a clear issue and we can reproduce it, then I would strongly advise opening an issue indeed!

Thanks for your input on this issue 👍

**Comment by elizaan36** [comment]
> If all is fine @elizaan36, I will ask to get this reviewed, and it should ship with the next release (WCPay 4.8.0 - scheduled to be released on Sept 28).

Looks good to me!

> Regarding the advanced settings, would you mind opening a low-priority issue to get the rewording sorted in [Woo admin](https://github.com/woocommerce/woocommerce)?

I can probably make that change, yeah.

> As for the feedback you've seen in the new user survey, could you share more details? I was just stating that it seemed unclear to me how one merchant could ever see the task "Reconnect WCPay" given that we seemed to require to be connected to even reach that Payments > Overview screen. But there might be other ways to reach that state that I'm not familiar with.

Will reach out on Slack with research details just to not hold this issue up, but one question - Don't these tasks also appear on the WooCommerce Home? Tasks in the extended task list are meant to appear in both places I thought.🤔

**Comment by jbordonado** [comment]
> Will reach out on Slack with research details just to not hold this issue up, but one question - Don't these tasks also appear on the WooCommerce Home? Tasks in the extended task list are meant to appear in both places I thought.🤔

I just double-checked and the `Update WooCommerce Payments business details` task also appears on WooCommerce Home, but not the `Reconnect WooCommerce Payments` one.

**Comment by jbordonado** [comment]
I updated the description and instructions to reflect the latest decisions.
You may now review the PR @fashuvo 👍

**Comment by elizaan36** [comment]
> I just double-checked and the Update WooCommerce Payments business details task also appears on WooCommerce Home, but not the Reconnect WooCommerce Payments one.

Thanks for checking on that @jbordonado. I think given the issue you brought up regarding the connection being required to even view the Payments Overview, we should make sure the "Reconnect WooCommerce Payments" task also appears on the WooCommerce Home screen. It would make the task behavior consistent as well, so that people don't wonder why some appear in one place and not the other.

As long as when the task is actioned upon (either completed or dismissed) it is completed/dismissed in both places.

**Comment by jbordonado** [comment]
Okay, I've made changes so that the "Reconnect" task also appears in WC > Home task list.

Good to note that instead of defining the field `content` to display the little text below the title, I'm now defining `additionalInfo` so that we can display the same text on both pages (WC > Home and Payments > Overview). Indeed, the rendering logic is slightly different between the tasks lists: the WC one is not using `content` while the WCPay one uses both `content` and `additionalInfo`, so I felt it was an easy way to achieve what we want without too much impact (there's an additional margin between the title and the text in Payments > Overview now, but since it is unlikely that we can reach that page when that task is active... I don't think it matters too much).

Also, since the tasks lists are built differently, I don't think we can have an "action button" nor an expandable card in WC > Home, so the task directly redirects to the Jetpack reconnect screen now.

Payments > Overview task list (notice the small margin, compared to previous screenshots I had shared previously):
image

WC > Home task list:
image

**Review by fashuvo — APPROVED** [review]
I see the `Enable secure checkout` prompt and the two tasks on Payments > Overview and WooCommerce > Home > Inbox. All tests passed.

LGTM 🚀

**Comment by rogermattic** [comment]
Hey @jbordonado! Sorry to drop this question here but I was wondering if you could tell me what does the `Update WooCommerce Payments business details` task do? I'm not sure what to do to replicate that to test it, and I wanted to investigate if it does not conflict with some other task that we're scoping to add to the payments task list (related to progressive onboarding). 

Feel free to kick me out of here, if you feel like this question doesn't belong here – happy to chat through this on Slack!

**Comment by jbordonado** [comment]
> Hey @jbordonado! Sorry to drop this question here but I was wondering if you could tell me what does the `Update WooCommerce Payments business details` task do? I'm not sure what to do to replicate that to test it, and I wanted to investigate if it does not conflict with some other task that we're scoping to add to the payments task list (related to progressive onboarding).
> 
> Feel free to kick me out of here, if you feel like this question doesn't belong here – happy to chat through this on Slack!

Hey, no problem.
I just checked, and it redirects to Stripe's login page. Once logged in, you end up on your Stripe Express account page

image

**Pull request merged**
